### PR TITLE
Conways front end

### PR DESCRIPTION
**Pull request by LucasUTNFRD**
## Features Implemented
- Basic game visualization with macroquad
- Game state management (running/paused)
- User interaction:
  - Space bar to pause/resume simulation
  - Left click to create living cells (when paused)
  - Right click to remove cells (when paused)
- Improved grid implementation with proper boundary checking
- Time-based updates for smooth simulation
- Initial glider pattern setup

### CODE REVIEW

File: src/conways.rs [hunk 1]
```diff
@@ -35,12 +33,14 @@ impl Grid {
     pub fn next_cell_generation(&mut self) {
         let mut new_grid = vec![vec![CellState::Dead; self.width]; self.height];
 
-        for y in 0..self.height {
-            for x in 0..self.width {
+        //for y in 0..self.height {
+        //    for x in 0..self.width {
```
Comment (dsocolobsky) [inline src/conways.rs]: Remove this commented code

File: src/main.rs [hunk 2]
```diff
@@ -1,9 +1,99 @@
...
+                    mouse_position().1 as usize / CELL_SIZE as usize,
+                );
+                self.grid.set(x, y, conways::CellState::Alive);
+            }
+            if is_mouse_button_down(MouseButton::Right) {
+                let (x, y) = (
+                    mouse_position().0 as usize / CELL_SIZE as usize,
+                    mouse_position().1 as usize / CELL_SIZE as usize,
+                );
+                self.grid.set(x, y, conways::CellState::Dead);
```
Comment (dsocolobsky) [inline src/main.rs]: This code is very similar I'm sure you could abstract it to a function such as `set_at_mouse_pos(state: CellState)` or similar.

File: src/conways.rs [hunk 3]
```diff
@@ -22,8 +20,8 @@ impl Grid {
         }
     }
 
-    pub fn set_alive(&mut self, x: usize, y: usize) {
-        self.grid[y][x] = CellState::Alive;
+    pub fn set(&mut self, x: usize, y: usize, state: CellState) {
+        self.grid[y][x] = state;
```
Comment (lima-limon-inc) [inline src/conways.rs:24]: I like this better, good idead ^_^

**Review by dsocolobsky — CHANGES_REQUESTED** [review]
Looks mostly good to me but I left some minor comments

File: src/conways.rs (hunk 1 above)
Comment (lima-limon-inc) [inline src/conways.rs]: I would remove this commented code. 

I always forget to remove it myself hahah 😅 ; I get you.

File: src/main.rs (hunk 2 above)
Comment (lima-limon-inc) [inline src/main.rs]: This is only a suggestion, and mind you, it is a "controversial" suggestion. But, I'll give it to you anyways; you decide if you like it or not ;).

Maybe, you could define a lambda function that does casting, and simply apply that function in all the cases. 

```rust
            let cast_to_usize = |num| num as usize / CELL_SIZE as usize;
            if is_mouse_button_down(MouseButton::Left) {
                let (x, y) = (
                    cast_to_usize(mouse_position().0),
                    cast_to_usize(mouse_position().1),
                );
                self.grid.set(x, y, conways::CellState::Alive);
            }
            if is_mouse_button_down(MouseButton::Right) {
                let (x, y) = (
                    cast_to_usize(mouse_position().0),
                    cast_to_usize(mouse_position().1),
                );
```

Like I said, controversial suggestion. Yet, the forbidden fruit tastes the sweetest

**Review by lima-limon-inc — APPROVED** [review]
Great Job Luki, looks amazing

File: src/conways.rs (hunk 1 above)
Comment (LucasUTNFRD) [reply src/conways.rs]: Done

File: src/main.rs (hunk 2 above)
Comment (LucasUTNFRD) [reply src/main.rs]: Done

File: src/conways.rs (hunk 3 above)
Comment (LucasUTNFRD) [reply src/conways.rs:24]: Thanks for the feedback Fabri!!!

File: src/conways.rs (hunk 1 above)
Comment (LucasUTNFRD) [reply src/conways.rs]: Done

File: src/main.rs (hunk 2 above)
Comment (LucasUTNFRD) [reply src/main.rs]: Thanks, i will take notes about this.

**Review by dsocolobsky — APPROVED** [review]

**Review by damiramirez — APPROVED** [review]
LGTM

**Review by lima-limon-inc — APPROVED** [review]
LGTM ^_^

**Pull request merged**